src/main/java/com/github/onsdigital/babbage/template/handlebars/helpers/markdown/util/MapTagReplacer.java

```diff
@@ -0,0 +1,126 @@
+    private static final String CHARSET = StandardCharsets.UTF_8.name();
+
+    /** the type of map to render - svg or png. */
+    public enum MapType {
+        PNG, SVG
+    }
+
+    private final String template;
+    private final ContentClient contentClient;
+    private final TemplateService templateService;
+    private final PooledHttpClient httpClient;
+    private final String rendererPath;
+
+    public MapTagReplacer(String path, String template, MapType mapType) {
+        this(path, template, ContentClient.getInstance(), TemplateService.getInstance(), HTTP_CLIENT, mapType);
+    }
+
+    public MapTagReplacer(String path, String template, ContentClient contentClient, TemplateService templateService, PooledHttpClient httpClient, MapType mapType) {
+        super(path);
+        this.template = template;
+        this.contentClient = contentClient;
+        this.templateService = templateService;
+        this.httpClient = httpClient;
+        if (mapType == MapType.PNG) {
+            this.rendererPath = RENDERER_PNG_PATH;
+        } else {
+            this.rendererPath = RENDERER_SVG_PATH;
+        }
+    }
+
+    /**
+     * Gets the pattern that this strategy is applied to.
+     *
+     * @return
+     */
+    @Override
+    public Pattern getPattern() {
+        return pattern;
+    }
+
+    /**
+     * The function that generates the replacement text for each match.
+     *
+     * @param matcher
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public String replace(Matcher matcher) throws IOException {
+
+        String tagPath = matcher.group(1);
+        String figureUri = resolveFigureUri(this.getPath(), Paths.get(tagPath)) + ".json";
+
+        try {
+            ContentResponse contentResponse = contentClient.getResource(figureUri);
+            String mapJson = contentResponse.getAsString();
+            String html = invokeMapRenderer(mapJson);
+            Map<String, Object> context = JsonUtil.toMap(mapJson);
+            String result = templateService.renderTemplate(template, context, Collections.singletonMap("mapHtml", html));
+            return result;
+        } catch (ResourceNotFoundException e) {
+            Log.buildDebug("Failed to find figure data for map.").addParameter("URL", figureUri).log();
+            return templateService.renderTemplate(figureNotFoundTemplate);
+        } catch (ContentReadException | MapRendererException e) {
+            Log.build("Failed rendering map, uri: " + figureUri + ", error: " + e, Level.ERROR).log();
+            return matcher.group();
+        }
+    }
+
+    private String invokeMapRenderer(String postBody) throws MapRendererException {
+        try (CloseableHttpResponse response = httpClient.sendPost(rendererPath, HEADERS, postBody, CHARSET)){
+            return IOUtils.toString(response.getEntity().getContent());
+        } catch (IOException e) {
```

Comment:
Log here please.

Reply:
that exception is caught and logged in the replace method, line 99:
```
        } catch (ContentReadException | MapRendererException e) {
```